query-parser/src/parse.rs

```diff
@@ -51,6 +51,25 @@ impl<I> Query<I>
     }
 }
 
+/// Generate a `satisfy_map` expression that matches a `PlainSymbol`
+/// value with the given name.
+///
+/// We do this rather than using `combine::token` so that we don't
+/// need to allocate a new `String` inside a `PlainSymbol` inside a `Value`
+/// just to match input.
+macro_rules! matches_plain_symbol {
```

Comment:
Very nice.  There might be an equivalent on the `tx-parser` side of the house, but I'll wait until I'm in that code again to file a more specific ticket.